### PR TITLE
SW-3536 Add planting season start/end months

### PR DESCRIPTION
**Pull request by sgrimm**
Add optional planting season start and end month fields to planting sites.
These will be used to determine the schedule for the site's observations.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1114** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1114" target="_blank">Graphite</a> 
    * **PR #1115** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1115" target="_blank">Graphite</a> 
      * **PR #1116** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1116" target="_blank">Graphite</a> 
        * **PR #1117** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1117" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1117?utm_source=stack-comment).

File: src/main/resources/db/migration/0151/V187__PlantingSiteDates.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE tracking.planting_sites ADD COLUMN planting_season_start_month INTEGER;
+ALTER TABLE tracking.planting_sites ADD COLUMN planting_season_end_month INTEGER;
+
+ALTER TABLE tracking.planting_sites ADD CONSTRAINT start_month_valid
+    CHECK (planting_season_start_month BETWEEN 1 AND 12);
+ALTER TABLE tracking.planting_sites ADD CONSTRAINT end_month_valid
+    CHECK (planting_season_end_month BETWEEN 1 AND 12);
```
Comment (karthikbtf) [inline src/main/resources/db/migration/0151/V187__PlantingSiteDates.sql:7]: if the end month is < start month, does that mean it is in the next year?

**Review by karthikbtf — APPROVED** [review]
Looks good 👍 One question on start/end months order.

File: src/main/resources/db/migration/0151/V187__PlantingSiteDates.sql (hunk 1 above)
Comment (sgrimm) [reply src/main/resources/db/migration/0151/V187__PlantingSiteDates.sql:7]: Right. Like if your planting season is in the summer and you're in the southern hemisphere, the season might start in December and end in February.

Comment (karthikbtf) [reply src/main/resources/db/migration/0151/V187__PlantingSiteDates.sql:7]: Thanks 👍 - makes sense.

**Pull request merged**